feat: support for EC keys in JWK sets and EC JWTs when using JWKS

shaunco opened the pull request:
test: new JWKS tests for RSA and EC
Fixes #719

FZambia commented on internal/jwtverify/token_verifier_jwt.go:
```diff
@@ -211,17 +211,34 @@ func (j *jwksManager) verify(token *jwt.Token, tokenVars map[string]any) error {
 	}
 
-	return verifier.Verify(token)
+	return errUnsupportedAlgorithm
```
Minor thing: can we move this to switch default?

FZambia reviewed (approved):
Many thanks, great addition, and special 👍 for tests. 

Looking at your commit message starting with `fix: ` - it seems you still think it's a bug 😈 😀

shaunco commented:
Fixed the PR title (😅) and moved that final return statement. I assume you will squash merge rather than wanting me to force push that last change, but let me know if I have that wrong.

FZambia commented:
Thanks a lot, this is great!

shaunco commented:
@FZambia - Thanks for getting this in! If you have a moment, a v5.0.5 release would be much appreciated!

FZambia commented:
@shaunco , Centrifugo is usually released not often than once in 3-4 weeks, except bug fix releases. Most probably v5.0.5 will be released till the end of this week - requires some time to prepare.

shaunco commented:
Thanks, I'll keep an eye out for it. We can't make use of it until the EC JWKS change is in a release.
